### PR TITLE
[WIP] Add dropdown component

### PR DESCRIPTION
**Pull request by lrojas94**

### CODE REVIEW

File: src/front/app/components/globals/Dropdown.js [hunk 1]
```diff
@@ -38,18 +44,27 @@ class Dropdown extends Component {
...
-      hide: { enabled: false },
-    },
   };
 
+  constructor(props) {
+    super(props);
+    this.handleDropdownButtonClick = this.handleDropdownButtonClick.bind(this);
+  }
+
+  state = {
```
Comment (RBrache21) [inline src/front/app/components/globals/Dropdown.js]: Shouldn't the state be inside the constructor?

**Comment by jrojas28** [comment]
@RBrache21  @lrojas94  A new commit has been implemented. It should feature a more complete, more useful Component. Feel free to check it and tell me what should be updated. If everything's in order, I'll proceed to merge.

File: src/front/app/components/globals/Dropdown.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { Manager, Reference, Popper } from 'react-popper';
```
Comment (lrojas94) [inline src/front/app/components/globals/Dropdown.js]: White space after external vs internal imports.

File: src/front/app/components/globals/Dropdown.js [hunk 3]
```diff
@@ -0,0 +1,143 @@
+import React, { Component } from 'react';
```
Comment (lrojas94) [inline src/front/app/components/globals/Dropdown.js]: Not seeing how this dropdown is gonna close when clicked outside, am I missing something?

File: src/front/app/components/globals/Dropdown.js [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        'dropdown-toggle': true,
+      });
+
+      return (
+        <div className={btnGroupClasses}>
+          <Button
+            type="primary"
+            className={btnClasses}
+            inputRef={ref}
+            onClick={onClick}
```
Comment (lrojas94) [inline src/front/app/components/globals/Dropdown.js]: You could call the method `openDropdown`. Not bad as onClick, but would be a bit more clear if you call it openDropdown.

File: src/front/app/components/globals/Dropdown.js [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+                    arrowProps,
+                  }) => (
+                    <div
+                      ref={ref}
+                      style={style}
+                      data-placement={placement}
+                      className={popperClasses}
+                    >
+                      {children}
+                      <div ref={arrowProps.ref} style={arrowProps.style} />
```
Comment (lrojas94) [inline src/front/app/components/globals/Dropdown.js]: Why tho? Dropdowns don't generally have an arrow pointing to the item, this is common in popups, but never seen it on DropDowns

Comment (jrojas28) [reply src/front/app/components/globals/Dropdown.js]: This was literally a mistake xD Consider it removed

File: src/front/app/components/globals/Dropdown.js [hunk 6]
```diff
@@ -104,7 +120,7 @@ class Dropdown extends Component {
     });
 
     return (
-      <div className="dropdown">
+      <div className="dropdown" ref={this.setWrapperRef}>
```
Comment (lrojas94) [inline src/front/app/components/globals/Dropdown.js]: `ref={ref => this.wrapperRef = ref}`. Quite cleaner imho.

**Comment by RBrache21** [comment]
it's completely working completely fine to me!

**Comment by RBrache21** [comment]
it's working completely fine to me!

**Pull request merged**